apps/frontend/src/pages/auth/components/sign-up-form/sign-up-form.tsx

```diff
@@ -24,31 +29,131 @@ const SignUpForm: React.FC<Properties> = ({ onSubmit }: Properties) => {
 	);
 
+	type InputType = {
+		icon: string;
+		type: "password" | "text";
+		value: string;
+	};
+
+	const [passwordData, setPasswordData] = useState<InputType>({
+		icon: slashEyeIcon,
+		type: "password",
+		value: "",
+	});
+
+	const [confirmPasswordData, setConfirmPasswordData] = useState<InputType>({
+		icon: slashEyeIcon,
+		type: "password",
+		value: "",
+	});
+
+	const [isSubmitEnabled, setIsSubmitEnabled] = useState(false);
+
+	const handleInputChange = useCallback(
+		(event_: React.BaseSyntheticEvent): void => {
+			const { name, value } = event_.target as HTMLInputElement;
+			if (name === "password") {
+				setPasswordData({ ...passwordData, value });
+			} else if (name === "confirmPassword") {
+				setConfirmPasswordData({ ...confirmPasswordData, value });
+			}
+		},
+		[passwordData, confirmPasswordData],
+	);
+
+	const toggleVisibilityPassword = useCallback(
+		(event_: React.BaseSyntheticEvent): void => {
+			event_.stopPropagation();
+			const imgAttribute = (event_.target as HTMLImageElement).dataset["name"];
+			if (imgAttribute == "password") {
+				const newData: InputType =
+					passwordData.icon === slashEyeIcon
+						? { ...passwordData, icon: eyeIcon, type: "text" }
+						: { ...passwordData, icon: slashEyeIcon, type: "password" };
+				setPasswordData(newData);
+			} else if (imgAttribute == "confirmPassword") {
+				const newData: InputType =
+					confirmPasswordData.icon === slashEyeIcon
+						? { ...confirmPasswordData, icon: eyeIcon, type: "text" }
+						: { ...confirmPasswordData, icon: slashEyeIcon, type: "password" };
+				setConfirmPasswordData(newData);
+			}
+		},
+		[passwordData, confirmPasswordData],
+	);
+
+	useEffect(() => {
+		const passwordsMatch = passwordData.value === confirmPasswordData.value;
+		setIsSubmitEnabled(passwordsMatch && passwordData.value !== "");
+	}, [passwordData.value, confirmPasswordData.value]);
+
 	return (
 		<>
-			<h3>Sign Up</h3>
+			<div className="logo-container">
+				<div className="circle circle-pink"></div>
+				<h1>LOGO</h1>
+			</div>
+			<h3 className="title-form">CREATE AN ACCOUNT</h3>
+			<span className="auth-info">
+				Already have an account? Go to
+				<Link className="link-info" to={AppRoute.SIGN_IN}>
+					{" "}
+					Log in
+				</Link>
+			</span>
 			<form onSubmit={handleFormSubmit}>
-				<p>
+				<div className="input-groups">
+					<Input
+						className="input-container"
+						control={control}
+						errors={errors}
+						label="Name"
+						name="name"
+						placeholder="name"
+						type="text"
+					/>
 					<Input
+						className="input-container"
 						control={control}
 						errors={errors}
 						label="Email"
 						name="email"
-						placeholder="Enter your email"
-						type="text"
+						placeholder="email"
+						type="email"
 					/>
-				</p>
-				<p>
 					<Input
+						className="input-container"
+						clickIcon={toggleVisibilityPassword}
 						control={control}
 						errors={errors}
+						hasIcon="true"
```

Comment:
```suggestion
						hasIcon={true}
```
Maybe you can write like this?